scipy/special/tests/test_basic.py

```diff
@@ -1319,6 +1319,17 @@ def test_comb(self):
         expected = 100891344545564193334812497256
         assert_equal(special.comb(100, 50, exact=True), expected)
 
+        with pytest.warns(
+            DeprecationWarning,
+            match="Non-integer arguments when exact is True are deprecated.",
```

Comment:
Will need to be adapted too, but you don't need to whole message. You can end with ".*" since the string is used as a regex.
```suggestion
            match="Non-integer arguments are deprecated when exact=True;.*",
```